### PR TITLE
analysis on CI for Flutter

### PR DESCRIPTION
**Pull request by dnfield**
## What does this PR accomplish?

Adds analysis to CI

## Did you add any dependencies?

package:pedantic as a dev dependency for the common set of lints it provides.

## How did you test the change?

It's new tests for CI.  Ran `flutter analyze` locally and fixed up analysis issues (unused imports).

### CODE REVIEW

File: client/flutter/lib/pages/travel_advice.dart [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import 'package:WHOFlutter/generated/l10n.dart';
```
Comment (myiremark) [inline client/flutter/lib/pages/travel_advice.dart]: looks like these were removed as unused imports, yes?

File: client/flutter/lib/pages/who_myth_busters.dart [hunk 2]
```diff
@@ -1,4 +1,3 @@
-import 'package:WHOFlutter/generated/l10n.dart';
```
Comment (myiremark) [inline client/flutter/lib/pages/who_myth_busters.dart]: same as above?

File: client/flutter/lib/pages/travel_advice.dart (hunk 1 above)
Comment (SamMousa) [reply client/flutter/lib/pages/travel_advice.dart]: Yes, OP describes it

File: .github/workflows/flutter-test.yml [hunk 3]
```diff
@@ -25,5 +25,7 @@ jobs:
       - run: echo "::add-path::$GITHUB_WORKSPACE/_flutter/bin"
       - run: flutter pub get
         working-directory: ./client/flutter
+      - run: flutter analyze
```
Comment (SamMousa) [inline .github/workflows/flutter-test.yml]:
```suggestion
      - run: flutter analyze --no-pub
```

File: .github/workflows/flutter-test.yml [hunk 4]
```diff
@@ -25,5 +25,7 @@ jobs:
...
+        working-directory: ./client/flutter
       - run: flutter test
```
Comment (SamMousa) [inline .github/workflows/flutter-test.yml]:
```suggestion
      - run: flutter test --no-pub
```
(I realize this is not part of analysis, but might as well add it in this PR)

**Comment by SamMousa** [comment]
@patniemeyer if this looks good to you i'll merge it

File: client/flutter/pubspec.yaml [hunk 5]
```diff
@@ -18,6 +18,7 @@ dependencies:
     sdk: flutter
 
 dev_dependencies:
+  pedantic: ^1.8.0
```
Comment (SamMousa) [inline client/flutter/pubspec.yaml]:
```suggestion
```

**Pull request merged**